### DISCUSSION TITLE
How create a custom parser?

### DISCUSSION THREAD

**Discussion by AxelBriche** [remarkjs/remark#1019] [Q&A] [answered]
Hello, I want create my own syntax `::my-component-name(foo: bar)::`, how can I create a custom parser with remark to convert this code into code like that?

`::my-component-name(foo: bar)::`

Becomes:

```
    {
        type: "component",
        name: "my-component-name",
        attributes: [
            { foo: bar }
        ]
    }
```

Thank you

**Comment by Murderlon** [comment, marked as answer]
[Extending markdown](https://github.com/micromark/micromark#extending-markdown) yourself is rarely a good idea. Instead I would recommend taking a look at [`remark-directive`](https://github.com/remarkjs/remark-directive). Or if you want components inside markdown: [MDX](https://github.com/mdx-js/mdx). Or if you want to load components with directives instead of JSX like [Nuxt content](https://content.nuxtjs.org/): [`remark-mdc`](https://www.npmjs.com/package/@docus/remark-mdc).
